components/admin_console/user_autocomplete_setting/user_autocomplete_setting.tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+
+import GenericUserProvider from 'components/suggestion/generic_user_provider.jsx';
+import Setting from 'components/admin_console/setting';
+import SuggestionBox from 'components/suggestion/suggestion_box';
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+
+type Props = {
+    id: string;
+    label: string;
+    placeholder: string;
+    helpText: string;
+    value: string;
+    onChange: (id: string, value: string) => void;
+    disabled: boolean;
+    actions: {
+        autocompleteUsers: (username: string) => (doDispatch: DispatchFunc) => Promise<UserProfile[]>;
+    };
+}
+
+const UserAutocompleteSetting = (props: Props) => {
+    const userSuggestionProviders = [new GenericUserProvider(props.actions.autocompleteUsers)];
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Since these are both passed into another component, they should be wrapped in a `useCallback` so that we're passing the same function into SuggestionBox instead of a new function each time

Reply:
Thanks for the guidance. Done